tests/views/unit.py

```diff
@@ -47,3 +50,91 @@ def test_get_units_ordered(rf, default, admin, test_get_units_po):
     request = create_api_request(rf, url=url, user=admin)
     response = view(request)
     assert response.status_code == 200
+
+
+@pytest.mark.django_db
+def test_submit_with_suggestion_and_comment(rf, default):
+    """Tests translation can be applied after suggestion is accepted."""
+    unit = Unit.objects.filter(suggestion__state='pending',
+                               state=UNTRANSLATED)[0]
+    sugg = Suggestion.objects.filter(unit=unit, state='pending')[0]
+    url = '/xhr/units/%d/' % unit.id
+    edited_target = "Edited %s" % sugg.target_f
+    comment = 'This is a comment!'
+    request = create_api_request(rf,
+                                 method='post',
+                                 url=url,
+                                 user=default)
+
+    request.POST = QueryDict(mutable=True)
+    for key, value in {
+        'state': False,
+        'target_f_0': edited_target,
+        'suggestion': sugg.id,
+        'comment': comment
+    }.items():
+        request.POST.appendlist(key, value)
+
+    response = submit(request, unit.id)
+    assert response.status_code == 200
+    accepted_suggestion = Suggestion.objects.get(id=sugg.id)
+    updated_unit = Unit.objects.get(id=unit.id)
+
+    assert accepted_suggestion.state == 'accepted'
+    assert str(updated_unit.target) == edited_target
+    assert get_model().objects \
+                      .for_model(accepted_suggestion) \
+                      .get().comment == comment
+
+
+@pytest.mark.django_db
+def test_accept_suggestion_with_comment(rf, admin):
+    """Tests suggestion can be accepted with a comment."""
+    unit = Unit.objects.filter(suggestion__state='pending',
+                               state=UNTRANSLATED)[0]
+    sugg = Suggestion.objects.filter(unit=unit, state='pending')[0]
+
+    url = '/xhr/units/%d/suggestions/%d/' % (unit.id, sugg.id)
+    comment = 'This is a comment!'
+    request = create_api_request(rf,
+                                 method='post',
+                                 url=url,
+                                 user=admin)
+
+    request.POST = QueryDict(mutable=True)
+    request.POST.appendlist('comment', comment)
+
+    response = accept_suggestion(request, unit.id, sugg.id)
+    assert response.status_code == 200
+    accepted_suggestion = Suggestion.objects.get(id=sugg.id)
+    updated_unit = Unit.objects.get(id=unit.id)
+
+    assert accepted_suggestion.state == 'accepted'
+    assert str(updated_unit.target) == str(sugg.target)
+    assert get_model().objects \
+                      .for_model(accepted_suggestion) \
+                      .get().comment == comment
+
+
+@pytest.mark.django_db
+def test_reject_suggestion_with_comment(rf, admin):
```

Comment:
can we make this test for `request_users` and make sure that the other users get the approp response, probs sim elsewhere